client/src/components/ChatStream.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { withStyles } from '@material-ui/styles';
+import uuid from 'react-uuid'
+import ChatTile from "./ChatTile";
+
+const styles = {
+    streamContainer: {
+        overflow: "auto",
+        "&::-webkit-scrollbar": {
+            display: "none"
+        }
+    }
+}
+
+class ChatStream extends React.Component {
+    constructor(props) {
+        super();
+        this.state = {
+            chats: [],
+            channel: null
+        }
+        this.messagesEndRef = React.createRef();
+    }
+
+    componentDidUpdate(prevProps) {
+        const { client } = this.props;
+        console.log(client);
+        if (!prevProps.client || client.channels[0] !== prevProps.client.channels[0]) {
+            this.updateTwitchClient();
+            this.setState({chats: []});
+        }
+    }
+
+    updateTwitchClient() {
+        const { client } = this.props;
+        client.connect();
+        client.on('message', (channel, user, message, self) => {
+            if (this.state.chats.length === 50)
+                this.state.chats.shift();
+            this.state.chats.push([user, message])
+            this.setState({chats: this.state.chats});
+            this.scrollToBottom();
+        });
+    }
+
+    scrollToBottom() {
+        this.messagesEndRef.current.scrollIntoView({ behavior: 'smooth' })
```

Comment:
When scrolling up in chat it shouldn't scroll into view, maybe showing how many new comments there are. Consider making an issue?

Reply:
Yes, this will be an issue in the future.

Reply:
Cool do you want me to make an issue for it?